server/calendar/helpers.js

```diff
@@ -10,55 +11,72 @@ const reUrl = /url\((.*?)\)/;
 const baseUrl = 'http://infozona.hr';
 
 module.exports = {
-  readDate, readTime,
-  readInfo, readPhotoUrl, readLink
+  readDate, readTime, readInfo,
+  readPhotoUrl, readLink, sanitizeHTML
 };
 
 function readDate($date) {
   let date = $date.text().trim();
-  if (reDate.test(date)) return date;
-  return formatDate(new Date());
+  date = normalizeDateString(date);
+  return reDate.test(date) ? date : formatDate(new Date());
 }
 
 function readTime($time) {
-  let [ time ] = $time.text().trim().split(/\s+/);
+  const [ time ] = $time.text().trim().split(/\s+/);
   return time.replace(/\./, ':');
 }
 
 function readInfo($info) {
-  let html = entities.decode($info.html());
-  let records = html.trim().split('<br>');
-  records = records.filter(r => r.length > 0);
-
-  let info = {};
-  records.forEach(record => {
-    let [key, val] = record.split(/:\s*/);
-    info[key] = val;
-  });
-  return info;
+  const html = entities.decode($info.html());
+  return html
+    .trim()
+    .split('<br>')
+    .filter(r => r.length > 0)
+    .reduce((info, record) => {
+      const [key, val] = record.split(/:\s*/);
+      info[key] = val;
+      return info
+    }, {});
 }
 
 function readPhotoUrl($photo) {
   if ($photo.length <= 0) return;
 
-  let style = $photo.attr('style');
-  let match = style.match(reUrl);
+  const match = $photo
+    .attr('style')
+    .match(reUrl);
```

Comment:
Not big fan of jQuery-esque formatting. If you really want to narrow it down make it oneliner.